### PR TITLE
[cyrolite] iP

### PR DESCRIPTION
**Pull request by cyrolite**
# **Scooby!**

> "The Key Is In Not Spending Time, But In Investing It." - Stephen R. Covey ([source](https://www.forbes.com/sites/katecooper/2022/07/03/meetings-the-key-is-in-not-spending-time-but-in-investing-it--stephen-r-covey/))

Feeling overwhelmed with deadlines, assignments, and commitments? We get it—life is hectic, and keeping track of everything can be a nightmare. That is why Scooby is here to help!

## Introducing Scooby!

Your **personalized** task manager, designed to help you stay on top of your work **effortlessly**—so you can focus on _what truly matters!_

Scooby is:

- text-based
- easy to learn
- user-friendly
- ~FAST~ SUPER FAST to use 👍 

## How to access Scooby?

1. Download the files [here](https://github.com/nus-cs2103-AY2425S2/ip/pull/108).
2. Go to Scooby.java and run the program:
```    
public static void main(String[] args) {
        Scooby sc = new Scooby();
        sc.run();
}
```
3. Add / delete tasks on your command!

It is that **SIMPLE**!

Features of Scooby:

- [X]  Add or Delete your tasks
- [X] List all of your tasks
- [X] Find any tasks to complete
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

### CODE REVIEW

**Review by chen-xiangrui — COMMENTED** [review]
LGTM! I like how the structure of your iP is very neat and tidy. The coding standards are also closely adhered to!

File: src/main/java/scooby/tasks/TaskList.java [hunk 1]
```diff
@@ -0,0 +1,259 @@
+package scooby.tasks;
+
+import java.util.ArrayList;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.File;
+import java.io.BufferedWriter;
+import java.io.FileWriter;
```
Comment (chen-xiangrui) [inline src/main/java/scooby/tasks/TaskList.java]: You could do some organising of the imported libraries and group those from the same ones (eg. java.io.*) together, with a blank line separating them from java.util.ArrayList

File: src/main/java/scooby/tasks/Task.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+     *
+     * @return the string representation of the task.
+     */
+    @Override
+    public String toString() {
+        return "[" + (isChecked ? "X" : " ") + "] " + desc;
+    }
+
+    /**
+     * Returns the raw representation of the String to be read from the txt file
```
Comment (chen-xiangrui) [inline src/main/java/scooby/tasks/Task.java:42]: The JavaDoc syntax is simple and easy to understand

File: src/test/java/scooby/ParserTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    @BeforeEach
+    void setUp() {
+        taskList = new TaskList(); // Using a real TaskList or simple stub
+        ui = new Ui("Scooby");
+        parser = new Parser(taskList, ui);
+    }
+
+    @Test
+    void testParseCommand_AddTask_EmptyDescription() {
```
Comment (chen-xiangrui) [inline src/test/java/scooby/ParserTest.java:26]: Useful test case to test the edge cases

**Review by chen-xiangrui — COMMENTED** [review]
LGTM! I like how the structure of your iP is very neat and tidy. The coding standards are also closely adhered to!
